content/hardware/10.mega/boards/giga-r1-wifi/tutorials/giga-debugging/giga-debugging.md

```diff
@@ -0,0 +1,184 @@
+
+## Goals
+
+The goals of this tutorial are:
+
+- learn to debug both cores on the GIGA R1 WiFi.
+- learn how to connect the GIGA R1 WiFi to the Segger J-Link debugger.
+- learn how to use the Ozone debugger software to debug an Arduino sketch.
+
+## Hardware & Software Needed
+
+- [Arduino IDE](https://www.arduino.cc/en/main/software)
+- [Arduino GIGA R1 WiFi](https://store.arduino.cc/products/giga-r1-wifi)
+- [GIGA R1 WiFi Board Package](/tutorials/giga-r1-wifi/giga-getting-started/)
+- [Segger J-Link](https://www.segger.com/products/debug-probes/j-link/)
+- [Ozone](https://www.segger.com/products/development-tools/ozone-j-link-debugger/)
+
+## Debugging
+
+Debugging your Arduino project allows you to dive deep into your code and troubleshoot as well as analyze the code execution. J-Link is a popular debugging tool that provides a direct connection between your computer and the Arduino board. By connecting it you can gain full access to the microcontroller's internal registers, memory, and variables. This is especially helpful when working on more complex projects where understanding the code execution flow is crucial. With J-Link and Ozone, you can step through the code line by line, allowing you to analyze why your code might break at a specific point.
+
+## Connection
+
+Connecting the J-link to your GIGA R1 WiFi is super easy because there are special pins for debugging labeled as **SWD** pins.
+
+![SWD Pins](./assets/jtag_pins.png)
+
+Your J-link should come with a cable included and if you look closer you will see that one side is marked with red, indicating the orientation of the cable. The red side of the cable should be on the same side as the VCC pin on the SWD pad. To know the exact orientation of your cable be sure to check the respective documentation or datasheet of your debugger but you can start with the orientation as shown below and if it doesn't work you can try placing it differently until you get a working connection. Once we start using the debugger software in the following steps you will know if the cable is placed correctly.
+
+## Software 
+
+### Setting up with the Arduino IDE
+
+First, if you haven't done it yet, install the [Arduino IDE](https://www.arduino.cc/en/software) and connect your GIGA R1 WiFi. When uploading a sketch to your Arduino board with the Arduino IDE, it will build an `.ELF` file of the sketch. We will need this file to debug in Ozone in the next steps. To easily find the file path of the `.ELF` file, we can enable the show verbose output option in the Arduino IDE. To do this, open up the preferences under **File > Preferences** in the Arduino IDE.
+
```

Comment:
If you have just installed the core you need to reset the IDE to make sure all options are visible to you. This includes the ability to upload to a second core.